### PR TITLE
Fix compilation error on CentOS 7 (gcc 4.8)

### PR DESCRIPTION
**Pull request by mateusztabaka**
libstdc++ that comes with gcc 4.8 doesn't
define `getline(basic_istream<char>&&, std::string&)`
even if it's part of the c++11 standard.
However we can still use the following:
`getline(basic_istream<char>&, std::string&)`.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by mateusztabaka** [comment]
@hzcyf what do you think?

**Comment by zihaomu** [comment]
Hi @mateusztabaka, thanks for your contribution!

I have a question about the code:
On my local computer (the compiler is llvm13), I found the 
```C++
std::getline(std::ifstream(video + "/device/interface"), uvcDev.name);
```
was parsed as the `getline(basic_istream<char>&, std::string&)` instead of `getline(basic_istream<char>&&, std::string&)`.

**Comment by hzcyf** [comment]
> @hzcyf what do you think?

@mateusztabaka Thanks for your test and contribution！Your modifications can improve code compatibility，and

**Comment by hzcyf** [comment]
> @hzcyf what do you think?

@mateusztabaka Thanks for your test and contribution！

I have tested this PR on ubuntu18.04 with GCC7.5， it‘s running well!
I think your modifications can improve the compatibility of our code. It should be merged.

Sorry for my bad English！

**Comment by mateusztabaka** [comment]
hi @zihaomu. That's interesting, I can see the same behavior that you mentioned with a simple snippet put into compiler explorer:  https://gcc.godbolt.org/z/3fcEbaz5M. Also similar behavior I see for gcc >= 6.1. However with no optimizations `getline(basic_istream<char>&&, std::string&)` is used. Summary in the table below:
<table>
<tr>
<th>compiler/optimizations</th><th>getline version used</th><th>link to CE</th>
</tr>
<tr>
<td>clang 13.0.1 with -O3</td><td>getline(basic_istream<char>&, std::string&)</td><td><a href="https://gcc.godbolt.org/z/3fcEbaz5M">https://gcc.godbolt.org/z/3fcEbaz5M</a></td>
</tr>
<tr>
<td>clang 13.0.1 with -O0</td><td>getline(basic_istream<char>&&, std::string&)</td><td><a href="https://gcc.godbolt.org/z/TeMrbohqx">https://gcc.godbolt.org/z/TeMrbohqx</a></td>
</tr>
<tr>
<td>gcc 12.1 with -O3</td><td>getline(basic_istream<char>&, std::string&)</td><td><a href="https://gcc.godbolt.org/z/EffabjYbs">https://gcc.godbolt.org/z/EffabjYbs</a></td>
</tr>
<tr>
<td>gcc 12.1 with -O0</td><td>getline(basic_istream<char>&&, std::string&)</td><td><a href="https://gcc.godbolt.org/z/dxxErxhE8">https://gcc.godbolt.org/z/dxxErxhE8</a></td>
</tr>
<tr>
<td>gcc 4.8.5</td><td>none, compilation error</td><td><a href="https://gcc.godbolt.org/z/fWYEcGcM5">https://gcc.godbolt.org/z/fWYEcGcM5</a></td>
</tr>

</table>

**Review by zihaomu — APPROVED** [review]
LGTM!👍 

I generated a successful test case on [GCC 4.8](https://gcc.godbolt.org/z/KjbWc156b).

**Comment by mateusztabaka** [comment]
Can this be merged or should we wait for someone else's approval (I can see that Alexander Alekhin is on vacation)?

**Comment by vpisarev** [comment]
@alalek, this fixes compile problems on CentOS, which is super-popular in some orgs. Can we merge it quickly?

**Review by asmorkalov — APPROVED** [review]

**Pull request merged**
